### PR TITLE
decoupling rptun ping and rptun_virtio_device, rptun ping and rptun.

### PR DESCRIPTION
**Pull request by wyr8899**
## Summary
Decoupling rptun ping and rptun_virtio_device, rptun ping and rptun.
ping is not only rptun support, so move it to public part.
## Impact
Decoupling ping and rptun, other transport layers can also call ping.
## Testing
Tested in sim vela, 86panel.

### CODE REVIEW

File: drivers/rptun/Kconfig [hunk 1]
```diff
@@ -38,11 +45,4 @@ config RPTUN_PM
 		goto RAM-retention mode, can't access from another CPU.
 		So, we provide this method to resolve this.
 
-config RPTUN_PING
```
Comment (raiden00pl) [inline drivers/rptun/Kconfig:41]: why moved ? doesn't this feature depend on CONFIG_RPTUN=y?

Comment (xiaoxiang781216) [reply drivers/rptun/Kconfig:41]: @raiden00pl we are restructure rptun to support non share memory transport(e.g. SPI/UART). The common part could be shared with them will be moved to driveers/rpmsg, ping will be part of the moving. After the reorganization, nuttx rptun/rpmsg driver framework will much like Linux kernel:
https://github.com/torvalds/linux/tree/master/drivers/rpmsg

Comment (raiden00pl) [reply drivers/rptun/Kconfig:41]: OK, that make sense

**Comment by raiden00pl** [comment]
CI fails for some reason

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**